xcm/pallet-xcm/src/lib.rs

```diff
@@ -448,9 +638,268 @@ pub mod pallet {
 			Self::deposit_event(Event::Attempted(outcome));
 			Ok(())
 		}
+
+		/// Extoll that a particular destination can be communicated with through a particular
+		/// version of XCM.
+		///
+		/// - `origin`: Must be Root.
+		/// - `location`: The destination that is being described.
+		/// - `xcm_version`: The latest version of XCM that `location` supports.
+		///
+		/// Errors:
+		/// - `BadLocation`: If the `location` cannot be expressed with the XCM version used by this
+		///   chain.
+		#[pallet::weight(100_000_000u64)]
+		pub fn force_xcm_version(
+			origin: OriginFor<T>,
+			location: Box<MultiLocation>,
+			xcm_version: XcmVersion,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			let location = *location;
+			SupportedVersion::<T>::insert(
+				XCM_VERSION,
+				LatestVersionedMultiLocation(&location),
+				xcm_version,
+			);
+			Self::deposit_event(Event::SupportedVersionChanged(location, xcm_version));
+			Ok(())
+		}
+
+		/// Set a safe XCM version (the version that XCM should be encoded with if the most recent
+		/// version a destination can accept is unknown).
+		///
+		/// - `origin`: Must be Root.
+		/// - `maybe_xcm_version`: The default XCM encoding version, or `None` to disable.
+		#[pallet::weight(100_000_000u64)]
+		pub fn force_default_xcm_version(
+			origin: OriginFor<T>,
+			maybe_xcm_version: Option<XcmVersion>,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			SafeXcmVersion::<T>::set(maybe_xcm_version);
+			Ok(())
+		}
+
+		/// Ask a location to notify us regarding their XCM version and any changes to it.
+		///
+		/// - `origin`: Must be Root.
+		/// - `location`: The location to which we should subscribe for XCM version notifications.
+		#[pallet::weight(100_000_000u64)]
+		pub fn force_subscribe_version_notify(
+			origin: OriginFor<T>,
+			location: Box<VersionedMultiLocation>,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			let location = (*location).try_into().map_err(|()| Error::<T>::BadLocation)?;
```

Comment:
I'm somewhat surprised that you need the explicit dereference here. Does the compiler not know how to auto-deref `location`?

Reply:
Oh, I'm guessing this is done due to type inference not being smart enough to recognize what the target type is for `try_into`.